src/macaron/repo_finder/commit_finder.py

```diff
@@ -0,0 +1,509 @@
+
+# The suffix separator exists for much the same purpose as the prefix separator: splitting the suffix into the actual
+# suffix, and the characters that join it to the version.
+# It optionally accepts:
+# One to two non-alphanumeric characters that are followed by either:
+# - A non-numeric character (positive lookahead).
+# - No character of any kind (negative lookahead).
+# E.g.
+# - '_'  of 'prefix_1.2.3_suffix'
+# - '..  of 'name-v-4-4-4..RELEASE'
+# - '#'  of 'v0.0.1#'
+SUFFIX_SEPARATOR = "(?P<suffix_sep>(?:[^0-9a-z]{1,2}(?:(?=[^0-9])|(?!.))))?"
+
+# The suffix optionally accepts:
+# A string that starts with an alphabetic character, and continues for one or more characters of any kind.
+SUFFIX = "(?P<suffix>[a-z].*)?"
+
+# If a version string has less parts than this number it will be padded with additional zeros to provide better matching
+# opportunities.
+# For this to be applied, the version string must not have any non-numeric parts.
+# E.g 1.2 (2) -> 1.2.0.0 (4), 1.2.RELEASE (3) -> 1.2.RELEASE (3), 1.DEV-5 (3) -> 1.DEV-5 (3)
+MAX_ZERO_DIGIT_EXTENSION = 4
+
+split_pattern = re.compile("[^0-9a-z]", flags=re.IGNORECASE)
+validation_pattern = re.compile("[0-9a-z]+$", flags=re.IGNORECASE)
+alphabetic_only_pattern = re.compile("[a-z]+$", flags=re.IGNORECASE)
+numeric_only_pattern = re.compile("[0-9]+$")
+versioned_string = re.compile("[a-z]+[0-9]+$", flags=re.IGNORECASE)  # e.g. RC1, M5, etc.
+
+
+def find_commit(git_obj: Git, purl: PackageURL) -> tuple[str, str]:
+    """Try to find the commit matching the passed PURL.
+
+    The PURL may have be a repository type, e.g. GitHub, in which case the commit might be in its version part.
+    Otherwise, the PURL should be a package manager type, e.g. Maven, in which case the commit must be found from
+    the artifact version.
+
+    Parameters
+    ----------
+    git_obj: Git
+        The repository.
+    purl: PackageURL
+        The PURL of the analysis target.
+
+    Returns
+    -------
+    tuple[str, str]
+        The branch name and digest as a tuple.
+    """
+    version = purl.version
+    if version is None:
+        logger.debug("Missing version for analysis target: %s", purl.name)
+        return "", ""
+
+    available_domains = [git_service.hostname for git_service in GIT_SERVICES if git_service.hostname]
+    domain = to_domain_from_known_purl_types(purl.type) or (purl.type if purl.type in available_domains else None)
+    if domain:
+        # PURL is a repository type.
+        return get_commit_from_purl(git_obj, version)
+    if purl.type in repo_finder_deps_dev.SUPPORTED_TYPES:
+        # PURL is a package manager type.
+        return get_commit_from_version(git_obj, purl.name or "", version)
+
+    logger.debug("Type of PURL is not supported for commit finding: %s", purl.type)
+    return "", ""
+
+
+def get_commit_from_purl(git_obj: Git, version: str) -> tuple[str, str]:
```

Comment:
Shouldn't the name of the function be 
```suggestion
def get_commit_from_purl_version(git_obj: Git, purl_version: str) -> tuple[str, str]:
```
instead? Because a full PURL is not passed in there, but only the version part of a PURL.